assets/app/scripts/filters/resources.js

```diff
@@ -144,4 +144,14 @@ angular.module('openshiftConsole')
       return (resource && resource.annotations && resource.annotations.provider) ||
         (resource && resource.metadata && resource.metadata.namespace);
     };
+  })
+  .filter('imageRepoReference', function(){
+    return function(objectRef, tag){
+      tag = tag || "latest";
+      var ns = objectRef.namespace || "";
+      ns = ns == "" ? ns : ns + "/";
+      var ref = ns + objectRef.name;
```

Comment:
is there any chance of name being undefined?  because appending it when its undefined will append "undefined"

If you aren't sure, do this to be safe:
```
var ref = ns + (objectRef.name || "");
```